inc/sitemaps/class-post-type-sitemap-provider.php

```diff
@@ -107,17 +100,12 @@ protected function get_classifier() {
 	/**
 	 * Get Home URL.
 	 *
-	 * This has been moved from the constructor because wp_rewrite is not available on plugins_loaded in multisite.
-	 * It will now be requested on need and not on initialization.
+	 * This wrapper method keeps up the compatibility with WPML. It returns "original home URL".
 	 *
 	 * @return string
 	 */
 	protected function get_home_url() {
-		if ( ! isset( self::$home_url ) ) {
-			self::$home_url = WPSEO_Utils::home_url();
-		}
-
-		return self::$home_url;
+		return WPSEO_Utils::home_url();
```

Comment:
Shouldn't we just deprecate this function and throw `WPSEO_Utils::home_url` as the alternative?

Reply:
WPML method _maybe_return_original_url_in_get_home_url_filter_ ( _classes/compatibility/wpseo/class-wpml-wpseo-xml-sitemaps-filter.php_ ) checks debug backtrace to force return of "original URL" for some methods:
```php
$places = array(
    array( 'WPSEO_Post_Type_Sitemap_Provider', 'get_home_url' ),
    array( 'WPSEO_Sitemaps_Router', 'get_base_url' ),
    array( 'WPSEO_Sitemaps_Renderer', '__construct' ),
);
```
If we remove method _get_home_url_ then it's possible that the classifier uses "wrong" base URL (_https&#58;//domain&#46;com/**en/**_ instead of _https&#58;//domain&#46;com/_).

@dgwatkins Could you help us with WPML compatibility?

I think that's the easiest to replace `WPSEO_Link_Type_Classifier( $this->get_home_url() )` with `WPSEO_Link_Type_Classifier( get_option( 'home' ) )`. As I know `get_option( 'home' )` should return "original URL".

Reply:
For backward compatibility, we are probably better off adding a new function name to the `$places` to check in the backtrace.

Does this sound good?

Reply:
Sounds sensible, means we can just refactor, right?

Reply:
Yes, and I would need to adjust WPML code.

I should be notified when this is merged to do this.
I assume I would have to use the trunk branch to test my WPML changes?

Reply:
@dgwatkins yes. I think if @stodorovic makes this change we'll merge this quite soon, after which you'd have a couple of weeks before this would be released.

Reply:
Since Yoast SEO 6.3 (PR #8524), there is the method _get_classifier_. @dgwatkins You can add this method (or even replace _get_home_url_ with _get_classifier_) in the list (`$places`). I just changed `WPSEO_Link_Type_Classifier( $this->get_home_url() );` to `WPSEO_Link_Type_Classifier( home_url() );` (same as on other places).